sympy/core/expr.py

```diff
@@ -808,7 +809,15 @@ def _eval_interval(self, x, a, b):
                 if isinstance(B, Limit):
                     raise NotImplementedError("Could not compute limit")
 
-        return B - A
+        value = B - A
+        a = a if a is not None else 0
```

Comment:
It a or b is None it is probably better not to consider singularities. It should suffice to return the `value` above.